cfnet/datasets.py

```diff
@@ -157,7 +157,9 @@ def __next__(self):
             if self.shuffle:
                 self.key = next(self.key_seq)
                 self.indices = jax.random.permutation(self.key, self.indices)
-            batch_data = [self.dataset[i] for i in self.indices[: self.batch_size]]
+            #batch_data = [self.dataset[i] for i in self.indices[: self.batch_size]]
+            batch_data = self.dataset[self.indices[: self.batch_size].tolist()]
+            batch_data = list(map(list, zip(*batch_data)))
```

Comment:
What is this line doing?

Reply:
zip and map are used to transpose the data, otherwise, the indexed result is incorrect for training. i.e. [[X_1, X_2, X_3], [Y_1, Y_2, Y_3]] instead of [[X_1,Y_1],[X_2,Y_2],[X_3,Y_3]].

Reply:
The indexed result is tuple, so there is no built-in transpose function.

Reply:
What we want is in fact a tuple `([X_1, X_2, X_3], [y_1, y_2, y_3])`.

In fact, most of the time `X` will be more than one dimension, so it should be `([[x_1, x_2, x_3]], [y_1])` if there is only one data in the batch.